### PR TITLE
Exclude .git, .venv, and other common non-source-code directories

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Can be cleaned up later, but IMO it's prob. including this (or something like it) in the next release

### CODE REVIEW

File: tests/test_get_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Parameters
+    ----------
+    monkeypatch
+        Pytest fixture, we use it to override ``PYTHONPATH``.
+
+    """
+    result = list(_get_notebooks("tests"))
+    assert Path("tests") / "data/clean_notebook.ipynb" in result
+
+    monkeypatch.setattr("nbqa.__main__.EXCLUDES", r"/(data)/")
```
Comment (girip11) [inline tests/test_get_notebooks.py:25]: This does not really test your EXCLUDES on code that will be shipped. Consider the scenario of having a bug in the `EXCLUDES` regex pattern. I would recommend to use `def test_get_notebooks(tmpdir: "LocalPath")`. Create `.git` dir inside this temp directory. Copy the notebook from `tests/data/clean_notebook.ipynb` to `tmpdir/.git`. You could do the same for many other directories like venv etc.

File: nbqa/__main__.py [hunk 2]
```diff
@@ -47,6 +48,12 @@
 VIRTUAL_ENVIRONMENTS_URL = (
     "https://realpython.com/python-virtual-environments-a-primer/"
 )
+EXCLUDES = (
+    r"/("
+    r"\.direnv|\.eggs|\.git|\.hg|\.ipynb_checkpoints|\.mypy_cache|\.nox|\.svn|\.tox|\.venv|"
```
Comment (girip11) [inline nbqa/__main__.py:53]: One way to simplify the regex pattern is to ignore the directories that start with `.`(hidden dirs). check if pattern like `r"/\.(\w|\s|-)+/"`

**Review by girip11 — CHANGES_REQUESTED** [review]

File: tests/test_get_notebooks.py (hunk 1 above)
Comment (MarcoGorelli) [reply tests/test_get_notebooks.py:25]: yes, that would be better, thanks!

File: nbqa/__main__.py (hunk 2 above)
Comment (MarcoGorelli) [reply nbqa/__main__.py:53]: true, but the fact that `black` doesn't do that makes me think we might run into something unexpected if we do that

Comment (girip11) [reply nbqa/__main__.py:53]: I am wondering why would someone want to run a notebook inside a hidden directory?

Comment (girip11) [reply nbqa/__main__.py:53]: If someone reports this as a bug/limitation, then we would know the scenario/usecase better right? That would be a better way to know about hidden dirs usage with notebooks.

Comment (s-weigand) [reply nbqa/__main__.py:53]: IMHO using an `--exclude` flag/option and glob patterns with [`fnmatch`](https://docs.python.org/3/library/fnmatch.html) as flake8 does, might be more robust and common, than regex.

**Pull request merged**